fabric-resource-conditions-api-v1/src/testmod/java/net/fabricmc/fabric/test/resource/conditions/DefaultResourceConditionsTest.java

```diff
@@ -0,0 +1,202 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.resource.conditions;
+
+import java.util.stream.Collectors;
+
+import com.mojang.serialization.JsonOps;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.registry.DynamicRegistryManager;
+import net.minecraft.registry.Registries;
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.RegistryWrapper;
+import net.minecraft.registry.tag.BiomeTags;
+import net.minecraft.registry.tag.BlockTags;
+import net.minecraft.registry.tag.TagKey;
+import net.minecraft.resource.featuretoggle.FeatureFlag;
+import net.minecraft.resource.featuretoggle.FeatureFlags;
+import net.minecraft.test.GameTest;
+import net.minecraft.test.TestContext;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.biome.BiomeKeys;
+
+import net.fabricmc.fabric.api.gametest.v1.FabricGameTest;
+import net.fabricmc.fabric.api.resource.conditions.v1.ResourceCondition;
+import net.fabricmc.fabric.api.resource.conditions.v1.ResourceConditions;
+import net.fabricmc.fabric.impl.resource.conditions.ResourceConditionsImpl;
+
+public class DefaultResourceConditionsTest {
+	private static final String TESTMOD_ID = "fabric-resource-conditions-api-v1-testmod";
+	private static final String API_MOD_ID = "fabric-resource-conditions-api-v1";
+	private static final String UNKNOWN_MOD_ID = "fabric-tiny-potato-api-v1";
+	private static final RegistryKey<? extends Registry<Object>> UNKNOWN_REGISTRY_KEY = RegistryKey.ofRegistry(new Identifier(TESTMOD_ID, "unknown_registry"));
+	private static final Identifier UNKNOWN_ENTRY_ID = new Identifier(TESTMOD_ID, "tiny_potato");
+
+	private void expectCondition(TestContext context, String name, ResourceCondition condition, boolean expected) {
+		RegistryWrapper.WrapperLookup registryLookup = context.getWorld().getRegistryManager();
+		boolean actual = condition.test(registryLookup);
+
+		if (actual != expected) {
+			throw new AssertionError("Test \"%s\" for condition %s failed; expected %s, got %s".formatted(name, condition.getType().id(), expected, actual));
+		}
+
+		// Test serialization
+		ResourceCondition.CODEC.encodeStart(JsonOps.INSTANCE, condition).getOrThrow(message -> new AssertionError("Could not serialize \"%s\": %s".formatted(name, message)));
+	}
+
+	@GameTest(templateName = FabricGameTest.EMPTY_STRUCTURE)
```

Comment:
Nit: All of these appear to be possible as junit unit tests, im not bothered at all though. The tests them selves look great 👍

Reply:
Note: There are some things like checking for DRM, where GameTest suits better.

Reply:
Yeah, thats fine. I have done some stuff with DRM in the unit test, would need to look into the specific case to see if its possible or not. Leave it as-is for now.